js/hw_3_js_react_1/src/components/Board/Board.jsx

```diff
@@ -0,0 +1,192 @@
+      setStart(false);
+    }
+    if (start && !pause) {
+      const intervalId = setInterval(() => {
+        setTime((prevTime) => prevTime - 1);
+      }, 1000);
+      return () => clearInterval(intervalId);
+    }
+  }, [start, time, gameOver, pause]);
+
+  useEffect(() => {
+    if (gameOver) {
+      return;
+    }
+    if (start && !gameOver && !pause) {
+      const intervalId = setTimeout(() => {
+        snakeMove();
+      }, speed)
+
+      return () => {
+        clearInterval(intervalId);
+      };
+    }
+  }, [direction, snake, gameOver, pause, start]);
+
+  function play() {
+    setSnake(initialSnakeBody);
+    setFood(generateNewFood(snake));
+    setDirection("ArrowUp");
+    setScore(0);
+    setGameOver(false);
+    setTime(initialTime);
+    setStart(true);
+  }
+
+  function changeDirection(e) {
+    if (e.key === "ArrowLeft" && direction !== "ArrowRight") {
+      setDirection("ArrowLeft");
+    } else if (e.key === "ArrowUp" && direction !== "ArrowDown") {
+      setDirection("ArrowUp");
+    } else if (e.key === "ArrowRight" && direction !== "ArrowLeft") {
+      setDirection("ArrowRight");
+    } else if (e.key === "ArrowDown" && direction !== "ArrowUp") {
+      setDirection("ArrowDown");
+    }
+    if (e.code === "Space") {
+      setPause(!pause);
+    }
+  }
+
+  function snakeMove() {
+      const snakeBodyBefore = [...snake];
+      const [snakeHeadX, snakeHeadY] = [...snake[0]];
+      const [foodX, foodY] = [...food[0]];
+      const pointForEat = 1;
+
+      let snakeBodyFirstElem = [snakeHeadX, snakeHeadY];
+
+      const movements = {
+        ArrowRight: ([x, y]) => [x + 1, y],
+        ArrowLeft: ([x, y]) => [x - 1, y],
+        ArrowUp: ([x, y]) => [x, y - 1],
+        ArrowDown: ([x, y]) => [x, y + 1],
+      };
+
+      const moveSnakeHead = movements[direction];
+      const snakeBodyFirstElemMove = moveSnakeHead(snakeBodyFirstElem);
+      const snakeBodyWithExtraTail = snakeBodyTail(
+        snakeBodyFirstElemMove,
+        snakeBodyBefore
+      );
+      const hasEatenFood = GetEat(
+        snakeBodyFirstElemMove,
+        foodX,
+        foodY
+      );
+      const snakeBody = snakeGrow(
+        hasEatenFood,
+        snakeBodyBefore,
+        snakeBodyFirstElemMove,
+        snakeBodyWithExtraTail
+      );
+
+      setGameOver(
+        getIsSnakeHeadOutsideBoard(...snakeBodyFirstElemMove) ||
+          getIsSnakeHeadOnBody([...snakeBodyFirstElemMove], snakeBody) ||
+          getIsTimeOut(time)
+      );
+
+      if (hasEatenFood) {
+        setScore((prevScore) => prevScore + pointForEat);
+        setFood(generateNewFood(snakeBody));
+      }
+
+      setSnake(snakeBody);
+
+  }
+
+  return (
+    <div>
+      <div className="mesh">
+        {board.map((row, indexX) => (
+          <div key={indexX} className="item">
+            {row.map((cell, indexY) => {
+              let snakeBody = snake.some((item) => item[0] === indexX && item[1] === indexY) && "snakeBody";
```

Comment:
This code needs refactoring - hard to maintain.

Also - you will get `<div className="false false">` which probably is not what you wanted.
https://www.npmjs.com/package/classnames check out this to help you handle classnames.